app/services/internal_analytics_service.py

```
@@ -0,0 +1,149 @@
+logger = logging.getLogger(__name__)
+
+class InternalAnalyticsService:
+    """
+    Lightweight analytics service using existing database infrastructure.
+    Tracks source detection usage and performance without external dependencies.
+    """
+    
+    def __init__(self):
+        self.db = get_db_instance()
+    
+    def track_source_detection_usage(self, 
```

Pylint (3.3.7)

[refactor] Too many arguments (10/5)

(R0913)